worlds/sc2/test/test_item_filtering.py

```diff
@@ -0,0 +1,83 @@
+"""
+Unit tests for item filtering like pool_filter.py
+"""
+
+from .test_base import Sc2SetupTestBase
+from ..item import item_tables, item_groups, item_names, item_parents
+from ..mission_tables import SC2Race
+
+class ItemFilterTests(Sc2SetupTestBase):
+    def test_excluding_all_barracks_units_excludes_infantry_upgrades(self) -> None:
+        return
+        # Note(mm): This test currently fails because of the circular logic around has_barracks_unit
+        # For an item to be filtered, logic must be satisfied without the item
+        # Terran logic includes (has_barracks_unit => infantry ups required)
+        # has_barracks_unit is only set to false after the items are culled
+        # Therefore, upgrades are always required by logic and will never be culled

```

Comment:
There was a step in the final pruning that removes unwanted W/A upgrades, this can't be done anytime sooner as you want to know what your units are